### PR TITLE
Made findByName endpoint 

### PR DESCRIPTION
**Pull request by James-Koval**
Made findByName endpoint (method and query), refactored and reformatted code connected with this logic, changed tests

### CODE REVIEW

File: src/main/java/ita/softserve/course_evaluation/controller/CourseController.java [hunk 1]
```diff
@@ -64,7 +77,7 @@
     @GetMapping
     public ResponseEntity<List<CourseDto>> getCourses() {
         return Objects.isNull(courseService.getAll()) ?
-                ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null) :
+                ResponseEntity.status(HttpStatus.NOT_FOUND).body(null) :
                 ResponseEntity.status(HttpStatus.OK).body(courseService.getAll());
     }
```
Comment (VovaProtsiv) [inline src/main/java/ita/softserve/course_evaluation/controller/CourseController.java:82]: Just return dto list. Don't return not found

File: src/main/java/ita/softserve/course_evaluation/controller/CourseController.java [hunk 2]
```diff
@@ -56,6 +56,19 @@
...
+            @ApiResponse(code = 200, message = HttpStatuses.OK, response = CourseDto.class),
+            @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+            @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
+    })
+    @GetMapping("name/{courseName}")
+    public ResponseEntity<List<CourseDto>> getCourseByName(@PathVariable String courseName) {
+        return Objects.isNull(courseService.getByName(courseName)) ?
+                ResponseEntity.status(HttpStatus.NOT_FOUND).body(null) :
+                ResponseEntity.status(HttpStatus.OK).body(courseService.getByName(courseName));
+    }
```
Comment (VovaProtsiv) [inline src/main/java/ita/softserve/course_evaluation/controller/CourseController.java:70]: Just return dto list. Don't return not found

File: src/main/java/ita/softserve/course_evaluation/entity/Course.java [hunk 3]
```diff
@@ -36,6 +37,7 @@
     private long id;
 
     @NotNull
+//    @Lob
     @Column(name = "course_Name", nullable = false)
```
Comment (VovaProtsiv) [inline src/main/java/ita/softserve/course_evaluation/entity/Course.java:41]: remove commented code

File: src/main/java/ita/softserve/course_evaluation/repository/CourseRepository.java [hunk 4]
```diff
@@ -2,21 +2,30 @@
...
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
 import java.util.Optional;
 
 @Repository
-public interface CourseRepository extends JpaRepository<Course, Long> {
+public interface CourseRepository extends JpaRepository<Course, Long>, CrudRepository<Course, Long> {
```
Comment (VovaProtsiv) [inline src/main/java/ita/softserve/course_evaluation/repository/CourseRepository.java:16]: JpaRepository extends CrudRepository

**Review by VovaProtsiv — CHANGES_REQUESTED** [review]

**Review by Arsen-Kushnir — APPROVED** [review]

**Review by VovaProtsiv — APPROVED** [review]

**Pull request merged**